### PR TITLE
fail build on missing application name

### LINKED ISSUE #437: kfdef application name missing doesnt fail `kfctl build`
```apiVersion: kfdef.apps.kubeflow.org/v1
kind: KfDef
metadata:
  namespace: kubeflow
spec:
  applications:
# other applications omitted
  - kustomizeConfig:
      repoRef:
        name: manifests
        path: stacks/ibm/application/spartakus
#    name: spartakus
  repos:
  - name: manifests
    uri: https://github.com/moficodes/manifests/archive/master.tar.gz
  version: master
```

The above kfdef builds fine. but apply would fail and the error message is completely unrelated. 
```
ERRO[0001] Error evaluating kustomization manifest for : accumulating resources: accumulating resources from '../../../../common/spartakus/overlays/application': open /common/spartakus/overlays/application: no such file or directory  filename="kustomize/kustomize.go:155"
Error: failed to apply:  (kubeflow.error): Code 500 with message: kfApp Apply failed for kustomize:  (kubeflow.error): Code 500 with message: error evaluating kustomization manifest for : accumulating resources: accumulating resources from '../../../../common/spartakus/overlays/application': open /common/spartakus/overlays/application: no such file or directory
```

We should add a check for application name or infer name from path.

### PR DESCRIPTION
**Pull request by moficodes**
fixes #437 

fixes formatting of struct

### CODE REVIEW

File: pkg/kfconfig/loaders/v1.go [hunk 1]
```diff
@@ -46,6 +46,12 @@ func (v V1) LoadKfConfig(def interface{}) (*kfconfig.KfConfig, error) {
 	config.ClusterName = kfdef.ClusterName
 	config.Spec.Version = kfdef.Spec.Version
 	for _, app := range kfdef.Spec.Applications {
+		if app.Name == "" {
+			return nil, &kfapis.KfError{
+				Code:    int(kfapis.INVALID_ARGUMENT),
+				Message: fmt.Sprintf("must have name for application"),
```
Comment (Tomcli) [inline pkg/kfconfig/loaders/v1.go:52]: maybe we can give a little bit for information in the message? like the index of the application array?

Comment (moficodes) [reply pkg/kfconfig/loaders/v1.go:52]: i thought of adding the repo path. but that might be empty. 

index makes more sense.

File: pkg/kfconfig/loaders/v1.go [hunk 2]
```diff
@@ -45,11 +45,11 @@ func (v V1) LoadKfConfig(def interface{}) (*kfconfig.KfConfig, error) {
 	config.Annotations = kfdef.Annotations
 	config.ClusterName = kfdef.ClusterName
 	config.Spec.Version = kfdef.Spec.Version
-	for _, app := range kfdef.Spec.Applications {
+	for i, app := range kfdef.Spec.Applications {
 		if app.Name == "" {
 			return nil, &kfapis.KfError{
 				Code:    int(kfapis.INVALID_ARGUMENT),
-				Message: fmt.Sprintf("must have name for application"),
+				Message: fmt.Sprintf("must have name for application. missing application name on application %d in kfdef", i),
```
Comment (Tomcli) [inline pkg/kfconfig/loaders/v1.go:52]: emphasize the number is the array index.
```suggestion
				Message: fmt.Sprintf("must have name for application. missing application name on application[%d] in kfdef", i),
```

File: pkg/kfconfig/types.go [hunk 3]
```diff
@@ -845,7 +845,7 @@ func (c *KfConfig) SetApplicationParameter(appName string, paramName string, val
 			"profiles":         KfAppsStackName,
 			"dex":              "dex",
 			// Spartakus is its own application because we want kfctl to be able to remove it.
-			"spartakus":                  "spartakus",
```
Comment (Jeffwan) [inline pkg/kfconfig/types.go:848]: minor: try to `go fmt` the file. seems dex and profiles are not justified.

**Review by Jeffwan — APPROVED** [review]

**Pull request merged**